hazelcast-jet-core/src/main/java/com/hazelcast/jet/Observable.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet;
+
+import com.hazelcast.jet.function.Observer;
+import com.hazelcast.jet.impl.observer.BlockingIteratorObserver;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.ringbuffer.Ringbuffer;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Represents a flowing sequence of events produced by one or more {@link
+ * Sinks#observable(String) observable sinks}.
+ * To observe the events, call {@link #addObserver
+ * jet.getObservable(name).addObserver(myObserver)}. The {@code Observable}
+ * is backed by a {@link Ringbuffer} and has a fixed capacity for storing
+ * messages. It supports reading by multiple subscribers, which will all
+ * observe the same sequence of messages. A new subscriber will start
+ * reading automatically from the oldest sequence available. Once the
+ * capacity is full, the oldest messages will be overwritten as new ones
+ * arrive.
+ * <p>
+ * Being backed by a {@link Ringbuffer}, an {@link Observable}'s capacity
+ * is basically the {@link Ringbuffer}'s capacity, which defaults to 10,000.
+ * So when using observables you don't need to worry about any data loss
+ * (not receiving some results) if you produce less than 10,000 of them (
+ * data loss only occurs if you produce them at a very high rate anyways).
+ * <p>
+ * If for some reason this default capacity is not suitable for your use case,
+ * then it can be configured on a case-by-case basis. For how to do this
+ * consult the IMDG Reference Manual, the "Configuring Ringbuffer Capacity"
+ * section. All the extra information you need is that the name of a
+ * {@link Ringbuffer} backing a certain {@link Observable} is
+ * a fixed prefix ("<strong>__jet.observables.</strong>") plus the
```

Comment:
we can't have this as part of public contract, since it's an implementation detail

Reply:
Ok, fixed it, have rewritten the whole Javadoc basically, pls. take a look.